eval/helm/main.py

```diff
@@ -0,0 +1,185 @@
+        if input_data.seed is not None:
+            L.seed_everything(input_data.seed)
+        logger.info("Using device: {}".format(fabric.device))
+        encoded = tokenizer.encode(
+            input_data.prompt, device=fabric.device
+        )
+        prompt_length = encoded.size(0)
+        max_returned_tokens = prompt_length + input_data.max_new_tokens
+        assert max_returned_tokens <= model.config.block_size, (
+            max_returned_tokens,
+            model.config.block_size,
+        )  # maximum rope cache length
+
+        t0 = time.perf_counter()
+        with fabric.init_tensor():
+            # set the max_seq_length to limit the memory usage to what we need
+            model.max_seq_length = max_returned_tokens
+
+        with fabric.init_tensor():
+            # enable the kv cache
+            model.set_kv_cache(batch_size=1)
+        tokens, logprobs, top_logprobs = generate(
+            model,
+            encoded,
+            max_returned_tokens,
+            temperature=input_data.temperature,
+            top_k=input_data.top_k,
+        )
+
+        t = time.perf_counter() - t0
+
+        model.clear_kv_cache()
+        if input_data.echo_prompt is False:
+            output = tokenizer.decode(tokens[prompt_length:])
+        else:
+            output = tokenizer.decode(tokens)
+        tokens_generated = tokens.size(0) - prompt_length
+        logger.info(
+            f"Time for inference: {t:.02f} sec total, {tokens_generated / t:.02f} tokens/sec"
+        )
+
+        logger.info(f"Memory used: {torch.cuda.max_memory_reserved() / 1e9:.02f} GB")
+        generated_tokens = []
+        for t, lp, tlp in zip(tokens, logprobs, top_logprobs):
+            idx, val = tlp
+            tok_str = tokenizer.processor.decode([idx])
+            token_tlp = {tok_str: val}
+            generated_tokens.append(
+                Token(text=tokenizer.decode(t), logprob=lp, top_logprob=token_tlp)
+            )
+        logprobs_sum = sum(logprobs)
+        # Process the input data here
+        return ProcessResponse(
+            text=output, tokens=generated_tokens, logprob=logprobs_sum, request_time=t
+        )
+
+    @app.post("/tokenize")
+    async def tokenize(input_data: TokenizeRequest) -> TokenizeResponse:
+        logger.info("Using device: {}".format(fabric.device))
+        t0 = time.perf_counter()
+        encoded = tokenizer.encode(input_data.text, device=fabric.device)
+        t = time.perf_counter() - t0
+        tokens = encoded.tolist()
+        return TokenizeResponse(tokens=tokens, request_time=t)
+
+    uvicorn.run(app, port=port)
+
+
+if __name__ == "__main__":
+    from jsonargparse import CLI
+
+    torch.set_float32_matmul_precision("high")
+    warnings.filterwarnings(
+        # Triggered internally at ../aten/src/ATen/EmptyTensor.cpp:31
+        "ignore",
+        message="ComplexHalf support is experimental and many operators don't support it yet",
+    )
+    CLI(main)
```

Comment:
Do this so that you always pass the argument name

```suggestion
    CLI(main, as_positional=False)
```